anvio/sequence.py

```diff
@@ -150,28 +152,68 @@ def __init__(self):
         self.mean_Q2Q3 = 0.0
 
 
-    def run(self, bam, split, ignore_orphans=False, max_coverage_depth=constants.max_depth_for_coverage):
-        coverage_profile = {}
+    def run(self, bam, contig_or_split, start=None, end=None, ignore_orphans=False, max_coverage_depth=constants.max_depth_for_coverage):
+        """Loop through the bam pileup and calculate coverage over a defined region of a contig or split
 
-        for pileupcolumn in bam.pileup(split.parent, split.start, split.end, 
-                                       ignore_orphans=ignore_orphans, max_depth=max_coverage_depth):
-            if pileupcolumn.pos < split.start or pileupcolumn.pos >= split.end:
-                continue
+        Parameters
+        ==========
+        bam : pysam.Samfile
+
+        contig_or_split : anvio.contigops.Split or anvio.contigops.Contig or str
+            If Split object is passed, and `start` or `end` are None, they are automatically set to
+            contig_or_split.start and contig_or_split.end. If str object is passed, it is assumed to
+            be a contig name
+
+        start : int
+            The index start of where coverage is calculated. Relative to the contig, even when
+            `contig_or_split` is a Split object. 
+
+        end : int
+            The index end of where coverage is calculated. Relative to the contig, even when
+            `contig_or_split` is a Split object.
+        """
+
+        if isinstance(contig_or_split, anvio.contigops.Split):
+            contig_name = contig_or_split.parent
+            if not start: start = contig_or_split.start
```

Comment:
for some reason I find this more readable in these cases:

``` python
start = contig_or_split.start if not start else start
```

Reply:
Ok I can change that